From hidave.darkstar@gmail.com  Tue May 12 09:29:37 2009
From: Dave Young <hidave.darkstar@gmail.com>
Date: Mon, 11 May 2009 14:17:45 +0800
Subject: kobject: make kset_create check kobject_set_name return  value
To: Greg KH <gregkh@suse.de>
Message-ID: <a8e1da0905102317n1be27da2lc9fd4dc1265b12ce@mail.gmail.com>


kset_create should check the kobject_set_name return value.

Add the return value checking code.

Signed-off-by: Dave Young <hidave.darkstar@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/lib/kobject.c
+++ b/lib/kobject.c
@@ -793,11 +793,16 @@ static struct kset *kset_create(const ch
 				struct kobject *parent_kobj)
 {
 	struct kset *kset;
+	int retval;
 
 	kset = kzalloc(sizeof(*kset), GFP_KERNEL);
 	if (!kset)
 		return NULL;
-	kobject_set_name(&kset->kobj, name);
+	retval = kobject_set_name(&kset->kobj, name);
+	if (retval) {
+		kfree(kset);
+		return NULL;
+	}
 	kset->uevent_ops = uevent_ops;
 	kset->kobj.parent = parent_kobj;
 
